ingestion/src/metadata/ingestion/source/tableau.py

```diff
@@ -83,25 +69,23 @@ class TableauSource(Source[Entity]):
         all_dashboard_details:
     """
 
-    config: TableauSourceConfig
+    config: WorkflowSource
     metadata_config: OpenMetadataServerConfig
     status: SourceStatus
 
     def __init__(
         self,
-        config: TableauSourceConfig,
+        config: WorkflowSource,
         metadata_config: OpenMetadataServerConfig,
     ):
         super().__init__()
         self.config = config
         self.metadata_config = metadata_config
         self.client = self.tableau_client()
         self.service = get_dashboard_service_or_create(
-            service_name=config.service_name,
+            service_name=config.serviceName,
             dashboard_service_type=DashboardServiceType.Tableau.name,
-            username=config.username,
-            password=config.password.get_secret_value() if config.password else None,
-            dashboard_url=config.server,
+            config=self.config.serviceConnection.__root__.dict(),
```

Comment:
is there a way we can refactor `get_dashboard_service_or_create` to not pass the dict here? In `get_database_service_or_create` we have a middle point, but I am not liking that a lot either.

If you can think of something elegant let's do that, otherwise we can open an issue and improve these functions in the future.

Thanks

Reply:
Yes - let's create an issue to resolve it for the later. Because the change will be needed in many connectors so let's open up a ticket and handle it later.